src/custom_widgets/toast.rs

```diff
@@ -0,0 +1,469 @@
+use std::fmt;
+use std::time::{Duration, Instant};
+
+use iced::advanced::layout::{self, Layout};
+use iced::advanced::overlay;
+use iced::advanced::renderer;
+use iced::advanced::widget::{self, Operation, Tree};
+use iced::advanced::{Clipboard, Shell, Widget};
+use iced::event::{self, Event};
+use iced::mouse;
+use iced::theme;
+use iced::widget::{button, column, container, horizontal_rule, horizontal_space, row, text};
+use iced::window;
+use iced::{Alignment, Element, Length, Point, Rectangle, Renderer, Size, Theme, Vector};
+
+pub const DEFAULT_TIMEOUT: u64 = 5;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub enum Status {
```

Comment:
I don't understand what the status is about. 
Could you add some code comments?
Doc comments ("///") are good, as then they pop-up in the IDE when hovering over a type...